src/platform/intel/ace/platform.c

```diff
@@ -80,18 +81,22 @@ static struct pm_notifier pm_state_notifier = {
 	.state_exit = cpu_notify_state_exit,
 };
 
+#if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+/* Value to be determined experimentaly */
+#define BASE_CPS_USAGE 10000
+#else
+#define BASE_CPS_USAGE (CLK_MAX_CPU_HZ / 1000)
+#endif
+
 /* Runs on the primary core only */
 int platform_init(struct sof *sof)
 {
 	int ret;
 
 	trace_point(TRACE_BOOT_PLATFORM_CLOCK);
 	platform_clock_init(sof);
-
-	/* Set DSP clock to MAX using KCPS API. Value should be lowered when KCPS API
-	 * for modules is implemented
-	 */
-	ret = core_kcps_adjust(cpu_get_id(), CLK_MAX_CPU_HZ / 1000);
+	kcps_budget_init();
```

Comment:
and this should be a separate change and be done earlier, before any of the kcps handling. The lock must be initialized

Reply:
yeap, moved to separate commit